docs/reference/release-notes/v2.28/index.md

```diff
@@ -0,0 +1,153 @@
+
+## Image plugin helpers
+
+This release adds new utility functions to help plugin authors add support for the [new Gatsby image component](https://github.com/gatsbyjs/gatsby/discussions/27950). If you maintain a source plugin, or if you want to help with one, take a look at [the RFC](https://github.com/gatsbyjs/gatsby/discussions/28241).
+
+## Removed experimental lazy page bundling
+
+In gatsby@2.27.0 we added [Experimental: Lazy page bundling](../v2.27/index.md#experimental-lazy-page-bundling-in-development) mode for `gatsby develop` that would delay compiling page templates until it was needed. While preliminary tests were very promising, we discovered few showstoppers that degraded development experience. [We decided to end the experiment](https://github.com/gatsbyjs/gatsby/discussions/28137#discussioncomment-138998) for now and shift our efforts to [Less aggressive cache invalidation](#less-aggressive-cache-invalidation).
+
+## Notable bugfixes
+
+- fix: do not ignore source file changes during recompilation [#28237](https://github.com/gatsbyjs/gatsby/pull/28237)
+- fix: invalidate page data when static query is added [#28349](https://github.com/gatsbyjs/gatsby/pull/28349)
+
+## Work in progress
+
+### Less aggressive cache invalidation
+
+Gatsby aggressively clears its cache, sometimes too aggressively. Here's a few examples:
+
+- You `npm install` a plugin, or update an existing
+- You change your `gatsby-node.js` and add a few log statements
+- You change your `siteMetadata` in `gatsby-config.js` to update your site's title
+
+In all of these caches, your cache is entirely cleared, which means that the next time you run `gatsby develop` the experience is slower than it needs to be. We'll be working on this to ensure that your first run, and every run thereafter, is as quick and speedy as you expect!
+
+## Experimental: Parallel data sourcing
+
+Plugin APIs in Gatsby run serially. Generally this what we want as most API calls are CPU/IO bound so things are fastest letting each plugin have the full undivided attention of your computer. But source plugins are often _network_ bound as they're hitting remote APIs and waiting for responses. We tried [changing the invocation of `sourceNodes` to parallel](https://github.com/gatsbyjs/gatsby/pull/28214) on a few sites with 4+ source plugins and saw a big speedup on sourcing (40%+) as they were no longer waiting on each other to start their API calls.
+
+This is a very "Your mileage may vary" situation — not all sites will notice any difference and also not all source plugins are network bound (gatsby-source-filesystem reads from the local machine). We're looking at finding better heuristics so that all sites are as fast as possible at data sourcing but in the meantime, if you have sites with multiple source plugins, this could be a big help.
+
+You can try it today using `gatsby@next` version and adding a flag to your `gatsby-config.js`:
```

Comment:
TODO: actually release `gatsby@next` (as https://github.com/gatsbyjs/gatsby/pull/28214 was already merged, but was too dangerous at this point to backport due to making changes to `apiRunner`)

Reply:
I'll publish `next`